src/main/java/co/rsk/bitcoinj/script/RedeemScriptParserFactory.java

```diff
@@ -26,57 +27,53 @@ public static RedeemScriptParser get(List<ScriptChunk> chunks) {
             return new NonStandardErpRedeemScriptParserHardcoded();
         }
 
-        ParseResult result = extractRedeemScriptFromChunks(chunks);
+        List<ScriptChunk> redeemScriptChunks = extractRedeemScriptFromChunks(chunks);
 
-        if (result == null) {
-            return new NonStandardErpRedeemScriptParserHardcoded();
-        }
-
-        if (FastBridgeRedeemScriptParser.isFastBridgeMultiSig(result.internalScript)) {
+        if (FastBridgeRedeemScriptParser.isFastBridgeMultiSig(redeemScriptChunks)) {
             logger.debug("[get] Return FastBridgeRedeemScriptParser");
             return new FastBridgeRedeemScriptParser(
-                result.internalScript
+                redeemScriptChunks
             );
         }
-        if (StandardRedeemScriptParser.isStandardMultiSig(result.internalScript)) {
+        if (StandardRedeemScriptParser.isStandardMultiSig(redeemScriptChunks)) {
             logger.debug("[get] Return StandardRedeemScriptParser");
             return new StandardRedeemScriptParser(
-                result.internalScript
+                redeemScriptChunks
             );
         }
-        if (P2shErpFederationRedeemScriptParser.isP2shErpFed(result.internalScript)) {
+        if (P2shErpFederationRedeemScriptParser.isP2shErpFed(redeemScriptChunks)) {
             logger.debug("[get] Return P2shErpFederationRedeemScriptParser");
             return new P2shErpFederationRedeemScriptParser(
-                result.internalScript
+                redeemScriptChunks
             );
         }
-        if (FastBridgeP2shErpRedeemScriptParser.isFastBridgeP2shErpFed(result.internalScript)) {
+        if (FastBridgeP2shErpRedeemScriptParser.isFastBridgeP2shErpFed(redeemScriptChunks)) {
             logger.debug("[get] Return FastBridgeP2shErpRedeemScriptParser");
             return new FastBridgeP2shErpRedeemScriptParser(
-                result.internalScript
+                redeemScriptChunks
             );
         }
-        if (ErpFederationRedeemScriptParser.isErpFed(result.internalScript)) {
+        if (ErpFederationRedeemScriptParser.isErpFed(redeemScriptChunks)) {
             logger.debug("[get] Return ErpFederationRedeemScriptParser");
             return new ErpFederationRedeemScriptParser(
-                result.internalScript
+                redeemScriptChunks
             );
         }
-        if (FastBridgeErpRedeemScriptParser.isFastBridgeErpFed(result.internalScript)) {
+        if (FastBridgeErpRedeemScriptParser.isFastBridgeErpFed(redeemScriptChunks)) {
             logger.debug("[get] Return FastBridgeErpRedeemScriptParser");
             return new FastBridgeErpRedeemScriptParser(
-                result.internalScript
+                redeemScriptChunks
             );
         }
 
         logger.debug("[get] Return NonStandardErpRedeemScriptParserHardcoded");
         return new NonStandardErpRedeemScriptParserHardcoded();
```

Comment:
not sure this is the best to do. If the redeem script is this special erp testnet one, it should have been detected at the beginning.

Reply:
This is addressed in the RedeemScriptParserFactory refactors task.